java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java

```diff
@@ -2075,6 +2076,10 @@ public int setCustomValues(User loggedInUser, Integer sid, Map<String, String> v
             }
         }
 
+        server.asMinionServer().ifPresent(minion -> {
+            MinionPillarManager.INSTANCE.generatePillar(minion);
```

Comment:
I know it's a bit more work, but I'd encourage creating an ad-hoc `MinionPillarGenerator` for this use case, or any change to custom info will entail the full regeneration of all pillar data, which can take a bit of time.

Reply:
Do you mean adding new pillar file or calling only MinionGeneralPillarGenerator?
Anyway, if I understand correctly, the write should go via MinionPillarManager.INSTANCE, right?
Is it OK to change MinionPillarManager to use named variables instead of list to hold MinionPillarFileManager instances and add MinionPillarManager.generatePillar with arguments to use only selected MinionPillarFileManager ?

Reply:
> Do you mean adding new pillar file or calling only MinionGeneralPillarGenerator?

I meant adding a new pillar file and related Generator class. Actually I made a copypaste mistake in that comment - meaning `MinionPillarGenerator` but actually writing `MinionGeneralPillarGenerator`, sorry for having confused you.

> Anyway, if I understand correctly, the write should go via MinionPillarManager.INSTANCE, right?

Only if you want to regenerate all pillar data, which IIUC should not be the case here.

> Is it OK to change MinionPillarManager to use named variables instead of list to hold MinionPillarFileManager instances and add MinionPillarManager.generatePillar with arguments to use only selected MinionPillarFileManager ?

It's not a bad idea FMPOV, the only thing I see is that, so far, the pattern has been to just instantiate a `new MinionPillarFileManager(new MySpecificPillarGenerator())` and call `updatePillarFile()`.

So I would welcome such change, at the condition it comes in its own separate commit (or even PR if you so prefer) and that all similar cases are changed in one shot (I see three of them apart from tests, see callers of `MinionPillarGenerator.generatePillarData()` as displayed by IntelliJ's `Navigate -> Call Hierarchy`).

![image](https://user-images.githubusercontent.com/250541/104381033-33bedb00-552c-11eb-8eda-0b9c1fb73b45.png)

Reply:
OK, I am working on it.
For completeness, ad-hoc MinionPillarFileManager instances do not work with unit tests which change pillar root.

Reply:
One more reason to go ahead with such a refactoring!